resizer.go

```diff
@@ -567,3 +607,106 @@ func getAngle(angle Angle) Angle {
 	}
 	return Angle(math.Min(float64(angle), 270))
 }
+
+func embedder(buf []byte, o Options) ([]byte, error) {
+	defer C.vips_thread_shutdown()
+
+	image, _, err := loadImage(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	image, err = vipsEmbed(image, o.Left, o.Top, o.Width, o.Height, o.Extend, o.Background)
+	if err != nil {
+		return nil, err
+	}
+
+	return saveImage(image, o)
+}
+
+func windowcropfixed(buf []byte, o Options) ([]byte, error) {
+	defer C.vips_thread_shutdown()
+
+	image, _, err := loadImage(buf)
+	if err != nil {
+		return nil, err
+	}
+	// ---------------------
+	// check if we need to crop the image
+	if o.Left != 0 || o.Top != 0 || int(image.Xsize) != o.AreaWidth || int(image.Ysize) != o.AreaHeight {
+		left, top, width, height := o.Left, o.Top, o.AreaWidth, o.AreaHeight
+		// first make sure the cropping area fits the image area
+		if o.Left < 0 {
+			left = 0
+			width += o.Left
+		}
+		if o.Top < 0 {
+			top = 0
+			height += o.Top
+		}
+		if left + width > int(image.Xsize) {
+			width = int(image.Xsize) - left
+		}
+		if top + height > int(image.Ysize) {
+			height = int(image.Ysize) - top
+		}
+		if width > 0 && height > 0 { // sanity check
+			image, err = vipsExtract(image, left, top, width, height)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// in case the window area was exceeding the image boundaries, embed the image in the window area
+		if int(image.Xsize) < o.AreaWidth || int(image.Ysize) < o.AreaHeight {
+			left, top := 0, 0		// where to put the cropped image
+			if o.Left < 0 {
+				left = -o.Left
+			}
+			if o.Top < 0 {
+				top = -o.Top
+			}
+			image, err = vipsEmbed(image, left, top, o.AreaWidth, o.AreaHeight, o.Extend, o.Background)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// now we have the cropped image of o.AreaWidth, o.AreaHeight size; resize it to match the target size
+	inWidth := int(image.Xsize)
+	inHeight := int(image.Ysize)
+	fmt.Printf("Cropped the image, the size is (%v, %v), output size should be (%v, %v)\n",
```

Comment:
Nope :) and other instances in this file

Reply:
I know, remove all the printfs :)